### DISCUSSION TITLE
How can I find out where types are missing in a large codebase ?

### DISCUSSION THREAD

**Discussion by williamdes** [vimeo/psalm#6003] [Q&A] [answered]
At phpMyAdmin we have `Psalm was able to infer types for 88.1163% of the codebase`
So I was wondering, how can I bring coverage to 100% ?

I did not find a way to print missing coverage files, do you have an idea how I can do this ?
A bit like phpunit html coverage, but a textual table would be okay

**Comment by orklah** [comment, marked as answer]
Running on level 1 should make Psalm emit Mixed* issues (like MixedAssignment) to tell you where the missing types are.

EDIT: of course, the obvious MissingReturnType, MissingParamType and MissingPropertyType should be looked into too

**Reply by williamdes** [reply]
Strangely I have none of these in the baseline file

**Reply by orklah** [reply]
can you share your xml config file?

**Reply by williamdes** [reply]
Here it is: https://github.com/phpmyadmin/phpmyadmin/blob/master/psalm.xml

**Reply by orklah** [reply]
You should remove this line: https://github.com/phpmyadmin/phpmyadmin/blob/master/psalm.xml#L4

It tells Psalm you don't want to do exactly what you ask

**Reply by williamdes** [reply]
Thank you !
